src/components/ModelingPageProvider.tsx

```diff
@@ -0,0 +1,175 @@
+import React, { useEffect, useMemo } from 'react'
```

Comment:
This file could be migrated into another files in the future.

It is useful at the moment since it is the largest component when loading the modeling page. It allows us to initialize a few things.

I'd say okay for the time being.